qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -0,0 +1,747 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Contains a (slow) Python simulator.
+
+It simulates a quantum circuit (an experiment) that has been compiled
+to run on the simulator. It is exponential in the number of qubits.
+
+The simulator is run using
+
+.. code-block:: python
+
+   BasicSimulator().run(run_input)
+
+Where the input is a :class:`.QuantumCircuit` object and the
+output is a :class:`.BasicProviderJob` object,
+which can later be queried for the Result object. The result will contain a 'memory' data
+field, which is a result of measurements for each shot.
+"""
+
+from __future__ import annotations
+
+import uuid
+import time
+import logging
+import warnings
+
+from collections import Counter
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import UnitaryGate
+from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
+from qiskit.providers import Provider
+from qiskit.providers.backend import BackendV2
+from qiskit.providers.models import BackendConfiguration
+from qiskit.providers.options import Options
+from qiskit.qobj import QasmQobj, QasmQobjConfig, QasmQobjExperiment
+from qiskit.result import Result
+from qiskit.transpiler import Target
+
+from .basic_provider_job import BasicProviderJob
+from .basic_provider_tools import single_gate_matrix
+from .basic_provider_tools import SINGLE_QUBIT_GATES
+from .basic_provider_tools import cx_gate_matrix
+from .basic_provider_tools import einsum_vecmul_index
+from .exceptions import BasicProviderError
+
+logger = logging.getLogger(__name__)
+
+
+class BasicSimulator(BackendV2):
+    """Python implementation of a basic (non-efficient) quantum simulator.
+
+    This implementation was originally based on the :class:`.BackendV1` interface,
+    and later migrated to follow :class:`.BackendV2`.
```

Comment:
I would just remove this line from the docstring since it's not something I think users really need to know as they'll only interact with the `BackendV2` API and `BasicSimulator` was not actually ever a `BackendV1` `BasicAer` was though ;) 
```suggestion
```